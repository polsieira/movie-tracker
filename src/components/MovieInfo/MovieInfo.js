import React from 'react';
import './MovieInfo.scss';
import { connect } from 'react-redux'

const MovieInfo = ({id, movies}) => {
  const movieId = parseInt(id.id)
  const selectedMovie = movies.find(movie => movie.id === movieId)
  const { title, release_date, vote_average, overview, backdrop_path } = selectedMovie
  const d = new Date(`${release_date}`);
  const date = `${d.getFullYear()}`;
    return (
      <section>
        <div>
          <figure className='movie-figure'>
            <figcaption>
              <h1>{`${title}(${date})`}</h1>
              <p>{vote_average}</p>
            </figcaption>
            <img className='movie-backdrop' src={`https://image.tmdb.org/t/p/original${backdrop_path}`} alt='movie backdrop' />
          </figure>
          <p>{overview}</p>
        </div>
      </section>
    )
}

const mapStateToProps = ({ movies }) => ({
  movies
})

export default connect(mapStateToProps)(MovieInfo); 