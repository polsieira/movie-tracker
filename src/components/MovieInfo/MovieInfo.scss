section {
  box-sizing: content-box;
  height: 100vh;
}

.movie-figure {
  margin: 0;
  width: 100%;
}

.movie-info-full {
  position: relative;
  padding: 0;
  margin: 0;
}


// This style at the moment is keeping the buttons from ever being active. If it is changed to a different container, the styles look worse but the buttons function.
.movie-figure:after {
  content: "";
  position: absolute;
  display: block;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0) linear-gradient(to bottom, rgba(0, 0, 0, 0) 0px, rgba(0, 0, 0, 0.6) 100%) repeat 0 0;
  z-index: 1;
}

figcaption {
  display: block;
  position: absolute;
  width: 100%;
  color: #fff;
  left: 0;
  bottom: 0;
  padding: 1em;
  font-weight: 700;
  z-index: 2;
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
}

.movie-nav {
  background-color: #1AAB8A;
  display: flex;
  height: 113px;
  justify-content: space-between;
}

.movie-backdrop {
  display: block;
  max-width: 100%;
  height: auto;
  margin: 0;
}

.movie-info-overview {
  position: absolute;
  padding: 10px;
}

.sign-in-movie,
.go-home {
  background:#1AAB8A;
  border: none;
  border-radius: 2px;
  color: #fff;
  cursor: pointer;
  height: 50px;
  font-size: 1em;
  outline: none;
  margin: 30px 30px 0px 30px;
  position: relative;
  transition: 800ms ease all;
  width: 110px;
  &:hover{
    background: #fff;
    color: #C3403F;
  }
  &:before,
  &:after {
    background: #1AAB8A;
    content: '';
    height: 2px;
    position: absolute;
    right: 0;
    top: 0;
    transition: 400ms ease all;
    width: 0;
  }
  &:after {
    bottom: 0;
    left: 0;
    right: inherit;
    top: inherit;
  }
  &:hover:before, 
  &:hover:after {
    transition: 800ms ease all;
    width: 100%;
  }
}