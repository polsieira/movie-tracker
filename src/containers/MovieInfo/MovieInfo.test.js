import { MovieInfo, mapStateToProps } from './MovieInfo';
import { shallow } from 'enzyme';
import React from 'react';


describe('MovieInfo', () => {
  describe('MovieInfo component', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<MovieInfo
        id={({ id: '5' })}
        movies={
          [{
            title: 'Wicked',
            release_date: '2017',
            vote_average: 6.5,
            popularity: 55.55,
            overview: 'Blah',
            backdrop_path: 'http://pic.pic',
            id: 5
          },
          {
            title: 'Joker',
            release_date: '2019',
            average_rating: 7.5,
            popularity: 68.55,
            id: 6
          }]}
      />)
    })

    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  });

  describe('mapStateToProps', () => {
    it('should return an object with the todos array', () => {
      // Setup
      const mockState = {
        movies: [{
          title: 'Wicked',
          release_date: '2017',
          average_rating: 6.5,
          popularity: 55.55,
          movie_id: 5
        },
        {
          title: 'Joker',
          release_date: '2019',
          average_rating: 7.5,
          popularity: 68.55,
          movie_id: 6
        }],
        user: {
          name: 'Pol',
          id: 2,
          isSignedIn: true
        }
      };
      const expected = {
        movies: [{
          title: 'Wicked',
          release_date: '2017',
          average_rating: 6.5,
          popularity: 55.55,
          movie_id: 5
        },
        {
          title: 'Joker',
          release_date: '2019',
          average_rating: 7.5,
          popularity: 68.55,
          movie_id: 6
        }]
      };

      // Execution
      const mappedProps = mapStateToProps(mockState);

      // Expectation
      expect(mappedProps).toEqual(expected);
    });
  });

});