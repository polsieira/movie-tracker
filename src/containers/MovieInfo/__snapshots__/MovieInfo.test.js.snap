// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo MovieInfo component should match a snapshot 1`] = `
<section>
  <div
    className="movie-info-full"
  >
    <div
      className="movie-nav"
    >
      <Link
        to="/"
      >
        <button
          className="go-home"
          type="button"
        >
          Home
        </button>
      </Link>
    </div>
    <figure
      className="movie-figure"
    >
      <figcaption>
        <h1
          className="movie-info-header"
        >
          Wicked(2016)
        </h1>
        <p
          className="movie-info-vote"
        >
          6.5
        </p>
      </figcaption>
      <img
        alt="movie backdrop"
        className="movie-backdrop"
        src="https://image.tmdb.org/t/p/originalhttp://pic.pic"
      />
    </figure>
    <p
      className="movie-info-overview"
    >
      Blah
    </p>
  </div>
</section>
`;
